shpc/main/modules/__init__.py

```diff
@@ -104,17 +104,15 @@ def _uninstall(self, module_dir, name, force=False):
         """
         Sub function, so we can pass more than one folder from uninstall
         """
-        if os.path.exists(module_dir) and not force:
-            msg = "%s, and all content below it? " % name
-            if utils.confirm_uninstall(msg, force):
-                self._cleanup(module_dir)
-                logger.info("%s and all subdirectories been removed." % name)
-
-        elif os.path.exists(module_dir) and force:
+        if os.path.exists(module_dir):
+            if not force:
+                msg = "%s, and all content below it? " % name
+                if not utils.confirm_uninstall(msg, force):
+                    return
```

Comment:
Sorry just missed one thing - what happened to:

```
self._cleanup(module_dir)
```
The subtle difference is cleaning up the module base:

https://github.com/singularityhub/singularity-hpc/blob/c9eca19b80d6f59a2ab216d8f93ac9cb2ccc2b3b/shpc/main/modules/__init__.py#L59-L63

Here a module directory might be the <container>/<version> however if all the versions are deleted, we'd want to move up one level and delete the container folder too. Take that one level higher, we clean up the modules directory.

Reply:
Ah, right ! Should this line https://github.com/singularityhub/singularity-hpc/blob/c9eca19b80d6f59a2ab216d8f93ac9cb2ccc2b3b/shpc/main/modules/__init__.py#L114 have been `self._cleanup(module_dir)` in the first place ?

Reply:
Yep, or at least now I can’t think of why the two cases should be different beyond my own oversight!